src/FluxCalculator.jl

```diff
@@ -91,4 +100,296 @@ function calculate_surface_air_density(atmos_sim::Interfacer.AtmosModelSimulatio
+    ::MoninObukhovScheme,
+    sim::Interfacer.SurfaceModelSimulation,
+    colidx::Fields.ColumnIndex,
+)
+    z0m = Interfacer.get_field(sim, Val(:roughness_momentum), colidx)
+    z0b = Interfacer.get_field(sim, Val(:roughness_buoyancy), colidx)
+    beta = Interfacer.get_field(sim, Val(:beta), colidx)
+    FT = eltype(z0m)
+    return (; z0b = z0b, z0m = z0m, Ch = FT(0), Cd = FT(0), beta = beta, gustiness = FT(1))
+end
+
+"""
+    surface_inputs(scheme::AbstractSurfaceFluxScheme, thermo_state_sfc, thermo_state_int, uₕ_int, z_int, z_sfc, z0b, z0m, Ch, Cd, beta, gustiness)
+
+Returns the inputs for the surface model simulation `sim`.
+"""
+function surface_inputs(
+    ::BulkScheme,
+    thermo_state_sfc,
+    thermo_state_int,
+    uₕ_int,
+    z_int,
+    z_sfc,
+    z0b,
+    z0m,
+    Ch,
+    Cd,
+    beta,
+    gustiness,
+)
+    FT = Spaces.undertype(axes(z_sfc))
+
+    # wrap state values
+    return @. SF.Coefficients(
+        SF.InteriorValues(z_int, uₕ_int, thermo_state_int), # state_in                              # state_sfc
+        SF.SurfaceValues(                                  # state_sfc
+            z_sfc,
+            StaticArrays.SVector(FT(0), FT(0)),
+            thermo_state_sfc,
+        ),
+        Cd,                                     # Cd
+        Ch,                                     # Ch
+        z0m,                                    # roughness_momentum
+        z0b,                                    # roughness_buoyancy
+        gustiness,                              # gustiness
+        beta,                                   # beta
+    )
+end
+function surface_inputs(
+    ::MoninObukhovScheme,
+    thermo_state_sfc,
+    thermo_state_int,
+    uₕ_int,
+    z_int,
+    z_sfc,
+    z0b,
+    z0m,
+    Ch,
+    Cd,
+    beta,
+    gustiness,
+)
+    FT = Spaces.undertype(axes(z_sfc))
+
+    # wrap state values
+
+    return @. SF.ValuesOnly(
+        SF.InteriorValues(z_int, uₕ_int, thermo_state_int), # state_in
+        SF.SurfaceValues(                                  # state_sfc
+            z_sfc,
+            StaticArrays.SVector(FT(0), FT(0)),
+            thermo_state_sfc,
+        ),
+        z0m,                                    # z0m
+        z0b,                                    # z0b
+        gustiness,                             # gustiness
+        beta,                                   # beta
+    )
+
+end
+
+"""
+    surface_thermo_state(sim::Interfacer.SurfaceModelSimulation, thermo_params::TD.Parameters.ThermodynamicsParameters, thermo_state_int, colidx::Fields.ColumnIndex)
+
+Returns the surface parameters for the surface model simulation `sim`.
+"""
+function surface_thermo_state(
+    sim::Interfacer.SurfaceModelSimulation,
+    thermo_params::TD.Parameters.ThermodynamicsParameters,
+    thermo_state_int,
+    colidx::Fields.ColumnIndex,
+)
+    @warn("Simulation " * Interfacer.name(sim) * " uses the default thermo (saturated) surface state", maxlog = 10)
```

Comment:
Are there cases that won't enter this method and so won't give a warning? Could you add a comment about this, maybe explaining the alternative thermo surface state?

Reply:
Sure.